src/lib.rs

```diff
@@ -417,6 +420,10 @@ impl IPTables {
             };
         }
 
-        Ok(output)
+        match output.status.code() {
```

Comment:
```rust
output.status.code().and_then(|i| if i == 0 { Ok(output) } else { Err(IPTError::BadExitStatus(i)) })
```

Reply:
AFAIK this is invalid: `and_then` return type `Option`.